typescript/common-resolvers/src/image.ts

```diff
@@ -125,7 +125,7 @@ export const getImageEntityFromMutationArgs = async (
       );
     }
 
-    await repository.upload.put(uploadTarget.uploadUrl, blob);
```

Comment:
I really think you should not do that this way. 
If you want, you can change the resolver for uploadTarget to return the right URL.
But having the tool do an upload to the download URL is a recipe for future disaster, it makes no sense.

Reply:
Yes we talked about that, I changed the parameters from being an URL to the key in the storage directly

Reply:
I'm not sure of the implications of doing that, but try to make sure that your solution will work well when we'll be able to plug several storage backends per dataset for example.

I feel that by having only the key be explicit, you make implicit assumption about the storage backend which is used, and now these assumptions will have to be shared somehow between the backend and the frontend.|

Will it work well if we say "This dataset is stored on the customer's S3 account in spexific bucket XYZ" ?

Reply:
The change came from the acknowledgement that the URL that was passed was systematically pruned to just keep the key. By digging a bit into that, there seems to be a semantic issue here: what would be an "uploadUrl" for s3? Is it the URL that points to the object? Then it's rather a download URL and cf your comment above, and you can't actually "put" to that URL. If it's a signedUrl for upload, then this function is trivial, which is okay, but it also already shows incompatibility with Supabase storage.

Reply:
I agree that the key is not sufficient to make the difference between let's say buckets / organizations inside a storage method, but there are even more parameters needed if we don't use signed URLs, like the API keys

Reply:
We could (should) do for upload just like we do for download: serve a redirection to a signedUrl that is generated during « getUploadTarget » using the S3 credentials. But that URL being a public presigned URL does not require credentials, temporary credentials are actually encoded in the presigned URL, that’s the point of those.

Of course this is not compatible with supabase for now, but for supabase backed datasets you just keep the current behavior and we have that 5mb limitation. 

URLs are a universal way to represent a resource, they give us flexibility for the future. But keys seems to couple things which rubs me in the wrong direction

Reply:
I'm totally fine with using signedUrls, but then wouldn't it just remove the need for a `repository.get` / `repository.put` since they'll both simply become a disguised `fetch` ?

Reply:
Yea indeed and that would be a good thing. BUT as of today for example out of the top of my head I know we’d still need repository.put in any case because to do a fetch put, you need an endpoint to do it, but due of the absence of upload presigned url in supabase, there are no endpoint to do fetch put to upload. So we created our own, but we can’t call our own endpoint from our own server side functions (e.g. to upload images contained in a zip file import). So we still need repository.put for that use case for example.

Reply:
If the interface to upload and download could be a simple external url instead of a function call, for all storage providers, then we could do without repository.put and repository.get. That would be nice. But for now we are blocked because of this supabase thing.

Reply:
Yes clearly, if we want to keep supporting Supabase we'll need these function; at least as long as Supabase don't implement upload signed URLs.

Reply:
After discussing with the others during the DSM, we found out that the main blocker for removing those `repository.upload.get` / `repository.upload.put` lies on the local service worker side rather than for the supabase solution. Actually we can query our API routes from server-side functions, but the service worker can't call query itself.

So I'll revert to expecting URLs in `repository.upload.get` / `repository.upload.put` that really make sense at least for server-side, meaning you do `put` and `get` on those URLs. About serving a redirection for the upload part, we discussed that as well, I thought that would be the best option since it preserves symmetry with respect to downloads but unfortunately I think we'd bump again in the 5Mb limit (cc @nacho-carnicero) since we'd send over the image in the body when putting on the API route, even if we don't actually use that body. I'm not exactly sure about how redirection works, but to my mind it's unlikely that the route ignores the body, so we'd rather provide an upload signedUrl directly when calling `getUploadTarget`.

Reply:
Yes, true, it's the SW the most problematic for this I forgot 👍 

For the 5Mb limit for uploads, I would try it just to be absolutely sure, but yes it makes sense.